### ISSUE TITLE
Streaming for unfunded accounts

### ISSUE THREAD

**Issue by TbLtzk** [stellar/go-horizon#332]
Consider this snippet (via js-stellar-sdk) for attaching to an account's effects stream:
```
    var futurePayments = Remote.getServer().effects().forAccount(accountId);
    var paymentsCloseHandle = futurePayments.stream({
            onmessage: function (effect) {
                effectHandler(effect); // process the new effect
            },
            onerror: function (error) {
                console.log(JSON.stringify(error));
            }
      });
```

Running this code for an unfunded account currently returns an error (see also #285).
This leaves no other option to client applications than polling horizon in regular intervals to get aware of the very first incoming transaction, the account-creation.

It would be desirable for many applications if the stream would not stop with an error, but keep connection open and notify if a `create_account ` operation is executed for the account of interest.

**Comment by TbLtzk** [comment]
The thrown error is not very helpful, btw, but this might also be an issue in js-stellar-sdk rather than horizon.
The above
`console.log(JSON.stringify(error));` 
currently yields
`{"isTrusted":true}`

**Comment by mvaneijk** [comment]
I agree that this functionality would be very handy, but an unfunded account just doesn't exist for Stellar. 

I don't think it would be a good idea to open streams for non existing accounts: If you can open a stream for non existing accounts, the server allows for opening unlimited streams.

- You could stream all the effects and wait for a `create_account` to the new account.
- Another solution could be a separate endpoint: `accounts/created` which streams newly created accounts.

**Comment by MikeFair** [comment]
If you are streaming for account creation, then you likely are in control of the account that is issuing the "create_account" operation.

Like @mvaneijk said, listen to the transaction stream for the funding account, when that stream sends a create_account event:
* take the created account id and add it to a global array of accounts that you are watching.  
* On a timer, loop over that global array and make sure the listener stream for each account is intact.
* Reinitiate the listener/streamer if it isn't.

This model will also help you deal with other errors that might interrupt the account streaming.
